crates/daphne-service-utils/src/compute_offload/mod.rs

```diff
@@ -0,0 +1,499 @@
+            }
+        });
+        builder.set_kdf_id(match kdf_id {
+            daphne::hpke::HpkeKdfId::HkdfSha256 => hpke_config::HpkeKdfId::HkdfSha256,
+            daphne::hpke::HpkeKdfId::NotImplemented(n) => {
+                unimplemented!("HpkeKdfId({n})")
+            }
+        });
+        builder.set_aead_id(match aead_id {
+            daphne::hpke::HpkeAeadId::Aes128Gcm => hpke_config::HpkeAeadId::Aes128Gcm,
+            daphne::hpke::HpkeAeadId::NotImplemented(n) => {
+                unimplemented!("HpkeAeadId({n})")
+            }
+        });
+        builder.set_public_key(public_key.as_slice());
+    }
+}
+
+impl CapnprotoPayloadDecode for HpkeConfig {
+    type Reader<'a> = hpke_config::Reader<'a>;
+
+    fn decode_from_reader(reader: Self::Reader<'_>) -> capnp::Result<Self>
+    where
+        Self: Sized,
+    {
+        Ok(Self {
+            id: reader.get_id(),
+            kem_id: match reader.get_kem_id()? {
+                hpke_config::HpkeKemId::P256HkdfSha256 => daphne::hpke::HpkeKemId::P256HkdfSha256,
+                hpke_config::HpkeKemId::X25519HkdfSha256 => {
+                    daphne::hpke::HpkeKemId::X25519HkdfSha256
+                }
+            },
+            kdf_id: match reader.get_kdf_id()? {
+                hpke_config::HpkeKdfId::HkdfSha256 => daphne::hpke::HpkeKdfId::HkdfSha256,
+            },
+            aead_id: match reader.get_aead_id()? {
+                hpke_config::HpkeAeadId::Aes128Gcm => daphne::hpke::HpkeAeadId::Aes128Gcm,
+            },
+            public_key: reader.get_public_key()?.into(),
+        })
+    }
+}
+
+impl CapnprotoPayloadEncode for Range<messages::Time> {
+    type Builder<'a> = time_range::Builder<'a>;
+
+    fn encode_to_builder(&self, mut builder: Self::Builder<'_>) {
+        builder.set_start(self.start);
+        builder.set_end(self.end);
+    }
+}
+
+impl CapnprotoPayloadDecode for Range<messages::Time> {
+    type Reader<'a> = time_range::Reader<'a>;
+
+    fn decode_from_reader(reader: Self::Reader<'_>) -> capnp::Result<Self>
+    where
+        Self: Sized,
+    {
+        Ok(reader.get_start()..reader.get_end())
+    }
+}
+
+impl CapnprotoPayloadEncode for PartialDapTaskConfigForReportInit<'_> {
+    type Builder<'a> = partial_dap_task_config::Builder<'a>;
+
+    fn encode_to_builder(&self, mut builder: Self::Builder<'_>) {
+        let PartialDapTaskConfigForReportInit {
+            not_after,
+            method_is_taskprov,
+            version,
+            vdaf,
+            vdaf_verify_key,
+        } = self;
+        builder.set_not_after(*not_after);
+        builder.set_method_is_taskprov(*method_is_taskprov);
+        builder.set_version((*version).into());
+        builder.set_vdaf(encode_vdaf_config(vdaf).as_slice().into());
+        {
+            let builder = builder.reborrow().init_vdaf_verify_key();
+            match vdaf_verify_key.as_ref() {
+                daphne::vdaf::VdafVerifyKey::L16(array) => {
```

Comment:
FYI: I'm not realizing we can remove this variant. I'll add a PR to follow up on this.

Reply:
https://github.com/cloudflare/daphne/pull/761